src/nano_contracts/serializer.ts

```diff
@@ -0,0 +1,216 @@
+   * @inner
+   */
+  fromInt(value: number): Buffer {
+    return signedIntToBytes(value, 4);
+  }
+
+  /**
+   * Serialize float value
+   *
+   * @param {value} Value to serialize
+   *
+   * @memberof Serializer
+   * @inner
+   */
+  fromFloat(value: number): Buffer {
+    return floatToBytes(value, 8);
+  }
+
+  /**
+   * Serialize boolean value
+   *
+   * @param {value} Value to serialize
+   *
+   * @memberof Serializer
+   * @inner
+   */
+  fromBool(value: boolean): Buffer {
+    if (value) {
+      return Buffer.from([1]);
+    } else {
+      return Buffer.from([0]);
+    }
+  }
+
+  /**
+   * Serialize a list of values
+   *
+   * @param {value} List of values to serialize
+   * @param {type} Type of the elements on the list
+   *
+   * @memberof Serializer
+   * @inner
+   */
+  fromList(value: any[], type: string): Buffer {
+    const ret: Buffer[] = [];
+    this.pushLenValue(ret, value.length);
+    for (const v of value) {
+      const serialized = this.serializeFromType(v, type);
+      ret.push(serialized);
+    }
+    return Buffer.concat(ret);
+  }
+
+  /**
+   * Serialize an optional value
+   *
+   * @param {isEmpty} If the optional has a value or not
+   * @param {value} Value to serialize (optional)
+   * @param {type} Type of the value to serialize (optional)
+   *
+   * @memberof Serializer
+   * @inner
+   */
+  fromOptional(isEmpty: boolean, value?: any, type?: string): Buffer {
+    // We are not supporting List optional for now
+    if (isEmpty) {
+      return Buffer.from([0]);
+    }
+
+    if (!value || !type) {
+      throw new Error('Missing value or type in non empty optional.');
+    }
+
+    const ret: Buffer[] = [];
+    ret.push(Buffer.from([1]));
+
+    const serialized = this.serializeFromType(value, type);
+    ret.push(serialized);
+    return Buffer.concat(ret);
+  }
+
+  /**
+   * Serialize a signed value
+   * We expect the value as a string separated by comma (,)
+   * with 3 elements (inputData, value, type)
+   *
+   * The serialization will be
+   * [len(serializedValue)][serializedValue][inputData]
+   *
+   * @param {signedValue} String value with inputData, value, and type separated by comma
+   *
+   * @memberof Serializer
+   * @inner
+   */
+  fromSigned(signedValue: string): Buffer {
+    const splittedValue = signedValue.split(',');
+    if (splittedValue.length !== 3) {
+      throw new Error('Signed data requires 3 parameters.');
+    }
+    // First value must be a Buffer but comes as hex
+    splittedValue[0] = hexToBuffer(splittedValue[0]);
+    if (splittedValue[2] === 'bytes') {
+      // If the result is expected as bytes, it will come here in the args as hex value
+      splittedValue[1] = hexToBuffer(splittedValue[1]);
+    } else if (splittedValue[2] === 'bool') {
+      // If the result is expected as boolean, it will come here as a string true/false
+      splittedValue[1] = splittedValue[1] === 'true';
+    }
+
+    const [inputData, value, type] = splittedValue;
```

Comment:
```suggestion
    let value: Buffer | string | bool;
    if (type === 'bytes') {
      // If the result is expected as bytes, it will come here in the args as hex value
      value = hexToBuffer(splittedValue[1]);
    } else if (type === 'bool') {
      // If the result is expected as boolean, it will come here as a string true/false
      value = splittedValue[1] === 'true';
    } else {
      // Assuming this is a string
      value = splittedValue[1];
    }
```

Reply:
Done https://github.com/HathorNetwork/hathor-wallet-lib/pull/604/commits/2f42d5432b729fce9cc7d7d09ebdb2ea280230c5